server/plugins/DemandUGens.cpp

```diff
@@ -589,6 +589,8 @@ void Duty_Ctor(Duty* unit) {
     unit->m_count = DEMANDINPUT(duty_dur) * SAMPLERATE;
     unit->m_prevout = DEMANDINPUT(duty_level);
     OUT0(0) = unit->m_prevout;
+    // calculate one sample of output.
+    unit->mCalcFunc(unit, 1);
```

Comment:
if i'm reading this correctly, this delays the entire output of the signal by 1 sample, which is a (slight) breaking change.

Reply:
Yes, and it is breaking another Unit Test : TestUGen_Duty

Mmmm. It's really more complicated than I thought...